Tools/TestWebKitAPI/Tests/WebCore/RegistrableDomain.cpp

```diff
@@ -99,4 +100,20 @@ TEST(RegistrableDomain, UncheckedCreateFromHost)
     ASSERT_EQ(dotWebkitDomainFromHost, webkitDomainFromString);
 }
 
+TEST(Site, EmptyMapKey)
+{
+    HashMap<Site, int> map;
+    map.set(Site(URL()), 1);
+    map.set(Site(URL("!@#$%^&*NotARealURL)(*&^"_s)), 2);
+    map.set(Site(SecurityOriginData(String(), String(), std::nullopt)), 3);
+    map.set(Site(String(), RegistrableDomain::fromRawString(String())), 4);
+    EXPECT_EQ(map.size(), 1u);
+    for (auto [site, value] : map) {
+        EXPECT_TRUE(site.isEmpty());
+        EXPECT_EQ(value, 4);
+    }
+    map.set(Site(String(), RegistrableDomain()), 5);
```

Comment:
I feel it's a bit counterintuitive that `Site(String(), RegistrableDomain::fromRawString(String()))` and `Site(String(), RegistrableDomain())` are not mapped to the same entry when `Site::isEmpty()` returns true for both.
If this is expected, I think we need to add a `EXPECT_TRUE(site.isEmpty());` here